### PR TITLE
[Requirements] Add google cloud dependencies to extra-requirements

### PR DESCRIPTION
**Pull request by gtopper**
Following #6146.

### CODE REVIEW

**Comment by tomerm-iguazio** [comment]
I was not able to run TestFeatureStoreGoogleBigQuery successfully, can you try to run these tests? @gtopper

**Comment by gtopper** [comment]
@tomerm-iguazio, I get one failure:
Screenshot 2024-08-20 at 19 54 59
But it's an unrelated issue in development, not something related to this PR.

**Review by tomerm-iguazio — APPROVED** [review]
Looks good:)

File: extras-requirements.txt [hunk 1]
```diff
@@ -34,6 +34,9 @@ plotly~=5.4, <5.12.0
 # To support python 11, google-cloud-bigquery requires grpcio>=1.49.1, which is incompatible with the grpcio version
 # required by frames (because it upgrades protobuf from 3.x to 4.x, breaking binary compatibility)
 google-cloud-bigquery[pandas, bqstorage]==3.14.1
+google-cloud-storage==2.14.0
```
Comment (tomerm-iguazio) [inline extras-requirements.txt:37]: please make sure your versions do not conflict with [Fix gcs recursive delete](https://github.com/mlrun/mlrun/pull/6193)

**Review by tomerm-iguazio — CHANGES_REQUESTED** [review]

**Review by tomerm-iguazio — APPROVED** [review]

**Pull request merged**